cdap-docs/examples-manual/source/examples/user-profiles.rst

```diff
@@ -0,0 +1,245 @@
+This is more precise, but it requires more book-keeping in the transaction
+system and thus can impact performance.
+
+Column-level conflict detection should be enabled if it is known that different
+transactions frequently modify different columns of the same row concurrently.
+
+UserProfiles Application
+------------------------
+
+This application uses:
+
+- a Stream ``events`` to receive events of user activity;
+- a Dataset ``profiles`` to store user profiles with conflict detection at either the row or column level;
+- a Dataset ``counters`` to count events by URL (this is not essential for the purpose of the example);
+- a Service ``UserProfileService`` to create, delete, and update profiles; and
+- a Flow ``ActivityFlow`` to count events and record the time of last activity for the users.
+
+The ``UserProfileService`` is a Service for creating and modifying user profiles. It has
+handlers to create, update, and retrieve user profiles.
+
+A script (``add-users.sh``) is used to populate the ``profiles`` Dataset. Two additional 
+scripts (``update-login.sh`` and ``send-events.sh``) are used to create a conflict by attempting
+to write to two different columns of the same row at the same time.
+
+
+Building and Starting
+=====================
+
+To observe conflict detection at both the row-level and column-level, you will need to modify 
+and build this example twice:
+
+- The first time, you will use row-level conflict detection, and see errors appearing in a log;
+- The second time, you will use column-level conflict detection and see the scripts complete successfully without errors.
+  
+Build the Application with Row-level Conflict Detection
+-------------------------------------------------------
+
+Before building the application, set the ``ConflictDetection`` appropriately in the class ``UserProfiles``:
+
+.. literalinclude:: /../../../cdap-examples/UserProfiles/src/main/java/co/cask/cdap/examples/profiles/UserProfiles.java
+      :language: java
+      :lines: 41-42
+      
+- The first time you build the application, set the ``tableProperties`` to ``ConflictDetection.ROW``. 
+
+- Build the example (as described `below <#building-an-example-application>`__).
+- Start CDAP, deploy and start the application and its component as described below in 
+  `Running CDAP Applications`_\ .
+  Make sure you start the Flow and Service as described below.
+- Once the application has been deployed and started, you can `run the example. <#running-the-example>`__
+- You should observe errors as described.
+
+Re-build the Application with Column-level Conflict Detection
+-------------------------------------------------------------
+
+- Stop the Application's Flow and Service (as described `below <#stopping-the-application>`__).
+- Delete the ``profiles`` Dataset, either through the CDAP Command Line Interface or
+  by making a ``curl`` call::
+
+    curl -w '\n' -v localhost:10000/v2/data/datasets/profiles -XDELETE
```

Comment:
It ends the curl call with a line feed, so that the return value doesn't have the command prompt run into the end. I have used them elsewhere, as it really improves the output when you cut and paste. As I use it to generate the output samples, I think we should give exactly the commands  that a user can use to get the same output.